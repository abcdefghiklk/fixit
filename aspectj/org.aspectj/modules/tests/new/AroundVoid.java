import org.aspectj.testing.Tester;

public class AroundVoid {
    public static void main(String[] args) {
        C c = new C();
        try {
            c.m1();
        } catch (RuntimeException exc) {
            Tester.event("caught RuntimeException");
        }
        c.m2();
        c.m3(true);
        c.m4(true);
        try {
            c.m5();
        } catch (ArithmeticException exc) {
            Tester.event("caught ArithmeticException");
        }
        c.m6();

        //Tester.printEvents();
        Tester.checkEventsFromFile("AroundVoid.out");
    }
}

class C {
    void m1() {
        throw new RuntimeException("m1");
    }

    void m2() {
        Tester.event("m2");
    }

    void m3(boolean test) {
        if (test) {
            return;
        } else {
            return;
        }
    }

    void m4(boolean test) {
        if (test) {
            return;
        } else {
            Tester.event("false");
        }
    }

    void m5() {
        while (true) {
            int x = 0;
            int y = 2/x;
            throw new ArithmeticException();
        }
    }

    void m6() {}
}

aspect A {
    Object around(): execution(void C.m*(..)) {
        Tester.event(thisJoinPoint.toShortString());
        return proceed();
    }
}
