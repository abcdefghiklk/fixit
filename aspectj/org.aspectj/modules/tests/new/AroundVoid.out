execution(C.m1())
caught RuntimeException
execution(C.m2())
m2
execution(C.m3(..))
execution(C.m4(..))
execution(C.m5())
caught ArithmeticException
execution(C.m6())
